source/unplugin/unplugin.civet

```diff
@@ -52,23 +52,30 @@ type CacheEntry
   result?: TransformResult
   promise?: Promise<void>
 
-civetExtension := /\.civet$/
+civetExtension := /\\.civet$/
 // Normally .jsx/.tsx extension should be present, but sometimes
 // (e.g. esbuild's alias feature) loads directly without resolve
-isCivetTranspiled := /(\.civet)(\.[jt]sx)?([?#].*)?$/
 postfixRE := /[?#].*$/s
 isWindows := os.platform() is 'win32'
 windowsSlashRE := /\\/g
-
-// Extracts query string and hash, and removes tsx/jsx extension
-function cleanCivetId(id: string): {id: string, postfix: string}
-  let postfix = ''
-  id = id
-  .replace postfixRE, (match) =>
+civetSuffix := '.civet'
+
+// Validate and extract the base Civet filename dynamically using outExt
+function validateAndExtractCivetBase(id: string, outputExtension: string): string | null
```

Comment:
```suggestion
function extractCivetFilename(id: string, outputExtension: string): string?
```

I think it's easier to just use `undefined` for the failure value. Also suggest a simpler name for the function.